### PR TITLE
Make Database Mutation Hooks Configurable

### PR DESCRIPTION
**Pull request by ArlonAntonius**
Makes the DatabaseMutation Hook configurable.
While testing noticed that MySQL changed the Tenant variable on the go, decided to replicate the model and set the tenant key manually.

### CODE REVIEW

File: src/Hooks/Database/Hooks/DatabaseMutation.php [hunk 1]
```diff
@@ -43,6 +44,15 @@ public function fires(): bool
         return Arr::has($this->mapping, get_class($this->event));
     }
 
+    public function for($event)
+    {
+        parent::for($event);
+
+        event(new ConfigureDatabaseMutation($event, $this));
```
Comment (luceos) [inline src/Hooks/Database/Hooks/DatabaseMutation.php]: Why under for() and not fire()? This looks out of place.

Comment (ArlonAntonius) [reply src/Hooks/Database/Hooks/DatabaseMutation.php]: Same as we did with the Migrations & Seeds hooks :)
Basically allows you to disable the hook before it even fires.

Comment (luceos) [reply src/Hooks/Database/Hooks/DatabaseMutation.php]: That's not how it was intended. You should use Tenancy\Pipeline\Events\Resolving for that.

Comment (ArlonAntonius) [reply src/Hooks/Database/Hooks/DatabaseMutation.php]: In that case, we can simply get rid of this.

Comment (ArlonAntonius) [reply src/Hooks/Database/Hooks/DatabaseMutation.php]: Fixed

**Pull request merged**